snooty/diagnostics.py

```diff
@@ -340,6 +340,57 @@ def __init__(
         self.entry = entry
 
 
+class UnknownTabset(Diagnostic):
+    severity = Diagnostic.Level.error
+
+    def __init__(
+        self,
+        tabset: str,
+        start: Union[int, Tuple[int, int]],
+        end: Union[None, int, Tuple[int, int]] = None,
+    ) -> None:
+        super().__init__(
+            f"tabset {tabset} is not defined in rstspec.toml", start, end,
+        )
+        self.tabset = tabset
+
+
+class UnknownTabID(Diagnostic):
+    severity = Diagnostic.Level.error
+
+    def __init__(
+        self,
+        tabid: str,
+        tabset: str,
+        reason: str,
+        start: Union[int, Tuple[int, int]],
+        end: Union[None, int, Tuple[int, int]] = None,
+    ) -> None:
+        super().__init__(
+            f"tab id {tabid} given in {tabset} tabset is unrecognized: {reason}",
+            start,
+            end,
+        )
+        self.tabid = tabid
+        self.tabset = tabset
+        self.reason = reason
+
+
+class TabMustBeDirective(Diagnostic):
+    severity = Diagnostic.Level.error
+
+    def __init__(
+        self,
+        tab_type: str,
+        start: Union[int, Tuple[int, int]],
+        end: Union[None, int, Tuple[int, int]] = None,
+    ) -> None:
+        super().__init__(
+            f"Tabs must be directives. This tab is of type {tab_type}", start, end
```

Comment:
As used, this diagnostic can also indicate non-tab directives. A better message might be `Tab sets may only contain tab directives, but found...`